include/onnxruntime/core/framework/allocator.h

```diff
@@ -72,6 +72,10 @@ struct OrtDevice {
   DeviceId device_id;
 };
 
+inline bool operator==(const OrtDevice& left, const OrtDevice& other) {
+  return left.Id() == other.Id() && left.MemType() == other.MemType() && left.Type() == other.Type();
```

Comment:
is memcmp more efficient?

Reply:
That would assume that object padding is always the same

---
In reply to: [310326739](https://github.com/microsoft/onnxruntime/pull/1553#discussion_r310326739) [](ancestors = 310326739)

Reply:
@yuslepukhin For the same compiler, same build, isn't it?

Reply:
Can the default implementation of operator== be used? 

---
In reply to: [310327045](https://github.com/microsoft/onnxruntime/pull/1553#discussion_r310327045) [](ancestors = 310327045,310326739)